jax/interpreters/partial_eval.py

```diff
@@ -362,9 +362,6 @@ def parents(self):
     else:
       return []
 
-  def ispure(self):
-    return self.pval.is_known()
-
   def full_lower(self):
```

Comment:
Actually, instead of removing `ispure` (since I think @gnecula only recently added it), can we adapt the logic in `full_lower` to use it, e.g.

```python
if self.ispure():
  return core.full_lower(self.pval.get_known())
else:
  return self
```

I'm kind of guessing this is something @gnecula may have had in mind, though he should correct me if I'm wrong!

(Perhaps also `get_known` should raise an error if it's not known, rather than returning a `None`...)

Reply:
I only changed the implementation for `ispure`. `get_known` returns None because that allowed the smallest change where previously we just fetched directly from the pair.

Reply:
Ah okay, thanks, I see now! Indeed `ispure` is super old and perhaps not needed. Let's kill it.

Reply:
Thanks for the correction!